### PR TITLE
OSSM-1916 Add go linter and fix all reported issues

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/controller/servicemesh/controlplane/deleter.go [hunk 1]
```diff
@@ -89,12 +92,15 @@ func (r *controlPlaneInstanceReconciler) Delete(ctx context.Context) error {
 			log.Info("Removed finalizer")
 			hacks.SkipReconciliationUntilCacheSynced(ctx, smcpNamespacedName)
 		} else if !apierrors.IsNotFound(err) {
-			r.EventRecorder.Event(instance, corev1.EventTypeWarning, eventReasonFailedRemovingFinalizer, fmt.Sprintf("Error occurred removing finalizer from service mesh: %s", err)) // TODO: this event probably isn't needed at all
-			return errors.Wrap(err, "Error removing ServiceMeshControlPlane finalizer")
+			// TODO: this event probably isn't needed at all
```
Comment (jwendell) [inline pkg/controller/servicemesh/controlplane/deleter.go:95]: worth to create an issue?

File: pkg/controller/servicemesh/webhooks/validation/memberroll.go [hunk 2]
```diff
@@ -185,14 +190,14 @@ func (v *MemberRollValidator) isUserAllowedToUpdatePodsInAllNamespaces(ctx conte
...
-			workerCtx := common.NewContextWithLogValues(ctx, "worker", i)
+		go func(workerNumber int) {
+			workerCtx := common.NewContextWithLogValues(ctx, "worker", workerNumber)
 			for ns := range in {
 				allowed, err := v.isUserAllowedToUpdatePods(common.NewContextWithLogValues(workerCtx, "namespace", ns), req, ns)
 				out <- result{namespace: ns, allowed: allowed, err: err}
 			}
 			wg.Done()
-		}()
+		}(i)
```
Comment (jwendell) [inline pkg/controller/servicemesh/webhooks/validation/memberroll.go:200]: interesting... I wonder what kind of obscure bugs could have happened here

File: tools/crd/main.go [hunk 3]
```diff
@@ -80,7 +77,7 @@ func processFile(fileName string, written sets.String) {
 								panic(fmt.Errorf("output file exists: %s", outFileName))
 							} else if os.IsNotExist(err) {
 								log.Info(fmt.Sprintf("writing CRD %s to %s", name, outFileName))
-								if err := ioutil.WriteFile(outFileName, []byte(object), 0664); err != nil {
+								if err := ioutil.WriteFile(outFileName, []byte(object), 0o664); err != nil {
```
Comment (jwendell) [inline tools/crd/main.go:80]: 🧐

**Review by jwendell — APPROVED** [review]
This is awesome, @luksa, thanks!

File: pkg/controller/servicemesh/webhooks/validation/memberroll.go (hunk 2 above)
Comment (luksa) [reply pkg/controller/servicemesh/webhooks/validation/memberroll.go:200]: Fortunately, this only caused the wrong worker number to be printed in the logs.

File: pkg/controller/servicemesh/controlplane/deleter.go (hunk 1 above)
Comment (luksa) [reply pkg/controller/servicemesh/controlplane/deleter.go:95]: I'll probably just remove the event the next time I make changes here.

**Pull request merged**